kore/src/Kore/Domain/Builtin.hs

```diff
@@ -657,6 +704,7 @@ data Builtin key child
     | BuiltinInt !InternalInt
     | BuiltinBool !InternalBool
     | BuiltinString !InternalString
+    | BuiltinBytes !InternalBytes
```

Comment:
Instead of adding this to `Builtin`, could we add an `InternalBytesF` constructor to `TermLikeF`? The `Builtin` type is a historical accident, and it's difficult to work with because it mashes many different things together.